src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/mgmt_iot_hub/swagger_iot_hub_resource_provider.json

```diff
@@ -0,0 +1,1913 @@
+        }
+      }
+    },
+    "IotHubProperties": {
+      "description": "The Iot Hub properties.",
+      "type": "object",
+      "properties": {
+        "state": {
+          "description": "IoT Hub state.",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Provision state.",
+          "type": "string"
+        },
+        "authorizationPolicies": {
+          "description": "The authorization rules.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedAccessSignatureAuthorizationRule"
+          }
+        },
+        "hostName": {
+          "description": "The name of the host.",
+          "type": "string"
+        },
+        "eventHubEndpoints": {
+          "description": "The event hub endpoint properties.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/EventHubProperties"
+          }
+        },
+        "storageEndpoints": {
+          "description": "The list of storage end points where files can be uploaded. Currently only one storage account can be configured.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/StorageEndpointProperties"
+          }
+        },
+        "messagingEndpoints": {
+          "description": "The list of messaging end points configured.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/MessagingEndpointProperties"
+          }
+        },
+        "enableFileUploadNotifications": {
+          "description": "The flag which indicates whether file upload notification should be enabled. This is optional at iot hub level. When enabled upload notifications will be available.",
+          "type": "boolean"
+        },
+        "cloudToDevice": {
+          "$ref": "#/definitions/CloudToDeviceProperties"
+        },
+        "comments": {
+          "description": "The comments.",
+          "type": "string"
+        },
+        "operationsMonitoringProperties": {
+          "$ref": "#/definitions/OperationsMonitoringProperties"
+        },
+        "features": {
+          "description": "The Capabilities/Features that need to be enabled for the Hub.",
+          "enum": [
+            "None",
+            "DeviceManagement"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "Capabilities",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "IotHubSkuInfo": {
+      "description": "The Sku related information for the hub.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the Sku.",
+          "enum": [
+            "F1",
+            "S1",
+            "S2",
+            "S3"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IotHubSku",
+            "modelAsString": true
```

Comment:
People (including me) have had misconceptions regarding what modelAsString truly means. At one point, the guidance was to interpret it as defining an inclusive list (the enum lists a set of values, but there could potentially be other valid values as well). This would have enabled the service to add new possible values without bumping the API version. This was an **incorrect** interpretation. 

The correct interpretation is that, whenever a client uses the API version defined in this swagger document, no values other than the ones listed in the enum can be returned.

As such, we *should* be able to use a choice list regardless of the modelAsString setting of the x-ms-enum extension. However, there are still cases where we have incorrect swagger definitions where this constraint is not followed (i.e. vm size).